git-rpc-server/git_rpc_server/server.py

```diff
@@ -0,0 +1,27 @@
+from jsonrpc import JSONRPCResponseManager, dispatcher
+from jsonrpc.exceptions import JSONRPCError
+import os
+from werkzeug.wrappers import Request, Response
+
+from git_rpc_server import config
+from git_rpc_server.methods import git, resources
+
+
+@Request.application
+def application(request):
+    """Listen for incoming requests on /jsonrpc"""
+    # We currently work on only one repo, the path to this
+    # repo is 'hardcoded' through the environment variable.
+    os.chdir(os.environ.get("MOUNT_PATH"))
+    # Add all methods
+    dispatcher.add_method(git.status, "git/status")
+    dispatcher.add_method(resources.disk_usage, "resources/diskUsage")
+    if config.RPC_SERVER_AUTH_TOKEN == request.headers.get(
+        config.RPC_SERVER_AUTH_HEADER_KEY
+    ):
+        response = JSONRPCResponseManager.handle(request.data, dispatcher)
+        return Response(response.json, mimetype="application/json")
+    return Response(
+        JSONRPCError(-32601, "Authentication is required.").json,
```

Comment:
According to the link below, json-rpc should always return status 200. The table in the link does not specify what you would do for 403 but for 404 (which in some cases is used instead of 403) they recommend that you use -32601. That is why I used this code here.

https://www.jsonrpc.org/historical/json-rpc-over-http.html#response-codes